### PR TITLE
Update the 1.1.5 branch to provide a native target and dependency.

### PR DESCRIPTION
**Pull request by phillipjohnston**
This is necessary for cross-compilation builds, where we want to link a test program on the build machine only. Without a separate dependnecy and library, there's no way to use this library with `native: true` targets.

---
This change is [Reviewable](https://reviewable.io/reviews/mesonbuild/cmocka/4)

### CODE REVIEW

**Comment by phillipjohnston** [comment]
I made a second commit which handles config.h properly for native and cross builds.

I don't love the duplication, but I don't currently have a better way to approach it. Open to suggestions if the maintainers have an idea.

**Comment by phillipjohnston** [comment]
Ping @sarum9in

**Comment by legeana** [comment]
Hi, I won't approve this as it is. Copy&paste build rules of this scale are not maintainable. I hardly follow what's going on there myself.
Few suggestions how to improve this and reduce build rule duplication:
- https://mesonbuild.com/Reference-manual.html#add_project_arguments has a `native` flag that allows to set separate flags for native and cross builds
- https://mesonbuild.com/Reference-manual.html#array-object and foreach and dicts can be used to loop over similar statements instead of duplication.

Please let me know what you think.

**Comment by phillipjohnston** [comment]
I will try the foreach approach to see if I can simplify it.

As for the "hardly follow what's going on there myself" - I didn't make the original file, so are you proposing that I need to improve the entire approach just to get accepted?

**Comment by phillipjohnston** [comment]
Ok @legeana I've updated based on your request and tested with a local cross-compilation build using CMocka as a subproject.

File: meson.build
Comment (legeana) [inline meson.build]: Nit: please name it `native` or `is_native` if `native` can't be used. Let's make it easier to read.

File: meson.build
Comment (legeana) [inline meson.build]: Should this be `gcc` for the cross build and `gcc_native` for the native build?

File: meson.build
Comment (legeana) [inline meson.build]: Is `dep_rt` actually used anywhere? I couldn't find it referenced.
I know you didn't introduce it, the reason I am asking however is that in the loop this variable is overwritten on the second iteration so this might have broken.

File: meson.build
Comment (legeana) [inline meson.build]: `cc` and `cc_native` definitions are rather trivial. I think you could inline them into the `cc_dict` and `cc_native_dict`.

**Review by legeana — CHANGES_REQUESTED** [review]
This version is so much better, thank you.

> are you proposing that I need to improve the entire approach just to get accepted

No. I am sorry for not being clear enough. What I meant is since your diff was huge it wasn't clear what you have actually changed for the native version of the build. Now that the code is shared it is much more understandable what your change actually is.

Overall it looks good, please take a look on my comments and once resolved I will release this version. Thank you for your patience.

File: meson.build
Comment (phillipjohnston) [reply meson.build]: Should just be `gcc` in both cases.

File: meson.build
Comment (phillipjohnston) [reply meson.build]: Doing a grep, I see no other uses in the repository. I will just remove this.

**Comment by phillipjohnston** [comment]
@legeana changes were made. Thank you for performing a thorough review, you have a keen eye.

File: meson.build
Comment (phillipjohnston) [reply meson.build]: Agreed here.

**Pull request merged**
